crates/net/p2p/src/error.rs

```diff
@@ -207,6 +202,18 @@ impl From<DatabaseError> for DownloadError {
     }
 }
 
+impl From<RequestError> for DownloadError {
+    fn from(error: RequestError) -> Self {
+        Self::RequestError(error)
+    }
+}
+
+impl From<ProviderError> for DownloadError {
+    fn from(error: ProviderError) -> Self {
+        Self::Provider(error)
+    }
+}
+
```

Comment:
same here as other pr, these don't have to be done manually, use https://jeltef.github.io/derive_more/derive_more/from.html

Reply:
Thanks for the review, this will make it much less verbose!

Reply:
exactly! and the great thing in turn with decreased verbosity, is that it's way easier to read the code that implements the functionality that sets this program apart from others